packages/nextjs/app/builders/0x04cdfb463600a29e8e93053b88daa11f54379907/page.tsx

```diff
@@ -0,0 +1,467 @@
+      ></div>
+      <div className={`transition-transform duration-300 group-hover:scale-110 group-hover:rotate-3 ${iconStyle}`}>
+        {icon}
+      </div>
+      <span className="text-xs font-semibold tracking-wide group-hover:translate-x-0.5 transition-transform duration-300">
+        {label}
+      </span>
+      <div
+        className={`absolute top-2 right-2 w-1 h-1 rounded-full opacity-0 group-hover:opacity-100 transition-opacity duration-500 ${isDark ? "bg-blue-400 shadow-[0_0_5px_rgba(96,165,250,0.8)]" : "bg-blue-600"}`}
+      ></div>
+    </div>
+  );
+}
+
+function AchievementCard({
+  title,
+  desc,
+  active = false,
+  icon,
+  isDark,
+}: {
+  title: string;
+  desc: string;
+  active?: boolean;
+  icon?: React.ReactNode;
+  isDark: boolean;
+}) {
+  const cardStyle = isDark
+    ? "bg-black/20 border-white/5 hover:bg-black/30"
+    : "bg-[#e2e8f0] border-slate-300 hover:border-blue-400 hover:shadow-lg";
+  const titleColor = isDark ? "text-white" : "text-slate-900";
+  const descColor = isDark ? "text-slate-400" : "text-slate-600";
+
+  return (
+    <div className={`flex items-start gap-4 p-4 rounded-2xl border transition-all text-left group ${cardStyle}`}>
+      <div
+        className={`mt-1 p-2 rounded-lg ${active ? "bg-green-500/10 text-green-500" : "bg-yellow-500/10 text-yellow-600"}`}
+      >
+        {icon ? icon : <CheckCircle2 size={20} />}
+      </div>
+      <div>
+        <h4 className={`font-semibold text-sm group-hover:text-blue-600 transition-colors ${titleColor}`}>{title}</h4>
+        <p className={`text-[11px] mt-1 leading-snug ${descColor}`}>{desc}</p>
+      </div>
+    </div>
+  );
+}
+
+const DharmaBuilder: NextPage = () => {
+  const { resolvedTheme } = useTheme();
+  const [mounted, setMounted] = useState(false);
+  const [copied, setCopied] = useState(false);
+  const [prices, setPrices] = useState<Record<string, { usd: number; usd_24h_change: number }> | null>(null);
+
+  const isDark = resolvedTheme === "dark";
+
+  useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  useEffect(() => {
+    const fetchPrices = async () => {
+      try {
+        const response = await fetch(
+          "https://api.coingecko.com/api/v3/simple/price?ids=bitcoin,ethereum,arbitrum,solana,optimism&vs_currencies=usd&include_24hr_change=true",
+        );
+        if (!response.ok) throw new Error("API Error");
+        const data = await response.json();
+        setPrices(data);
+      } catch {
+        setPrices({
+          bitcoin: { usd: 96500, usd_24h_change: 2.5 },
+          ethereum: { usd: 3650, usd_24h_change: -1.2 },
+          arbitrum: { usd: 1.85, usd_24h_change: 5.4 },
+          solana: { usd: 195, usd_24h_change: 1.1 },
+          optimism: { usd: 2.1, usd_24h_change: -0.5 },
+        });
+      }
+    };
+
+    fetchPrices();
+    const interval = setInterval(fetchPrices, 60000);
+    return () => clearInterval(interval);
+  }, []);
```

Comment:
Consider using tanstack/react-query for fetching external data. It facilitates handling the fetch state and keeps the code cleaner.